arangod/Agency/MoveShard.cpp

```diff
@@ -710,10 +767,16 @@ JOB_STATUS MoveShard::pendingLeader() {
                            trx.add(VPackValue(planPath));
                            {
                              VPackArrayBuilder guard(&trx);
-                             for (VPackSlice srv : VPackArrayIterator(plan)) {
-                               if (!srv.isEqualString(_from)) {
-                                 trx.add(srv);
+                             if (plan.isArray()) {
+                               for (VPackSlice srv : VPackArrayIterator(plan)) {
+                                 if (!srv.isEqualString(_from)) {
+                                   trx.add(srv);
+                                 }
                                }
+                             } else {
+                               LOG_TOPIC("37714", WARN, Logger::SUPERVISION)
+                                 << "failed to iterate over planned servers for "
+                                 << _shard << " or one of its clones";
```

Comment:
You want to set `failed` to `true` here.

Reply:
yes

Reply:
and return